runtime/service/src/main/java/org/apache/polaris/service/auth/JWTBroker.java

```diff
@@ -156,16 +142,18 @@ public TokenResponse generateFromClientSecrets(
     if (principal.isEmpty()) {
       return new TokenResponse(OAuthTokenErrorResponse.Error.unauthorized_client);
     }
-    String tokenString = generateTokenString(clientId, scope, principal.get().getId());
+    String tokenString =
+        generateTokenString(principal.get().getName(), principal.get().getId(), clientId, scope);
     return new TokenResponse(
         tokenString, TokenType.ACCESS_TOKEN.getValue(), maxTokenGenerationInSeconds);
   }
 
-  private String generateTokenString(String clientId, String scope, Long principalId) {
+  private String generateTokenString(
+      String principalName, long principalId, String clientId, String scope) {
     Instant now = Instant.now();
     return JWT.create()
         .withIssuer(ISSUER_KEY)
```

Comment:
side note: it might be worth setting `aud` to the realm ID and later validating it (for follow up). WDYT?